doc/source/whatsnew/v0.21.0.txt

```diff
@@ -405,6 +405,7 @@ Reshaping
 - Bug in :func:`crosstab` where passing two ``Series`` with the same name raised a ``KeyError`` (:issue:`13279`)
 - :func:`Series.argmin`, :func:`Series.argmax`, and their counterparts on ``DataFrame`` and groupby objects work correctly with floating point data that contains infinite values (:issue:`13595`).
 - Bug in :func:`unique` where checking a tuple of strings raised a ``TypeError`` (:issue:`17108`)
+- Bug in :func:`concat` which would randomly determine the order of the index along the common dimension (:issue:`17344`)
```

Comment:
this is not very imformative as this is a very specific case that you are fixing ( it makes it seem like this is a very common / huge problem)